candi/tools/update_kubernetes_patchversions.sh

```diff
@@ -28,36 +30,55 @@ function check_requirements() {
 # update_version_map VERSION PATCH
 function update_version_map() {
   local NEW_DIGEST
-  yq -i e ".k8s.\"${1}\".patch = ${2}" /deckhouse/candi/version_map.yml
+  yq -i e ".k8s.\"${1}\".patch = ${2}" ../version_map.yml
   # Kube-proxy
-  if yq -e e "select(.k8s.\"${1}\".controlPlane | has(\"kubeProxy\"))" /deckhouse/candi/version_map.yml >/dev/null 2>/dev/null; then
+  if yq -e e "select(.k8s.\"${1}\".controlPlane | has(\"kubeProxy\"))" ../version_map.yml >/dev/null 2>/dev/null; then
     NEW_DIGEST="$(crane digest "registry.k8s.io/kube-proxy:v${1}.${2}")"
-    yq -i e ".k8s.\"${1}\".controlPlane.kubeProxy = \"${NEW_DIGEST}\"" /deckhouse/candi/version_map.yml
+    yq -i e ".k8s.\"${1}\".controlPlane.kubeProxy = \"${NEW_DIGEST}\"" ../version_map.yml
   fi
   # Kube-scheduler
   NEW_DIGEST=""
-  if yq -e e "select(.k8s.\"${1}\".controlPlane | has(\"kubeScheduler\"))" /deckhouse/candi/version_map.yml >/dev/null 2>/dev/null; then
+  if yq -e e "select(.k8s.\"${1}\".controlPlane | has(\"kubeScheduler\"))" ../version_map.yml >/dev/null 2>/dev/null; then
     NEW_DIGEST="$(crane digest "registry.k8s.io/kube-scheduler:v${1}.${2}")"
-    yq -i e ".k8s.\"${1}\".controlPlane.kubeScheduler = \"${NEW_DIGEST}\"" /deckhouse/candi/version_map.yml
+    yq -i e ".k8s.\"${1}\".controlPlane.kubeScheduler = \"${NEW_DIGEST}\"" ../version_map.yml
   fi
 }
 
 check_requirements
 
-TMPDIR="$(mktemp -du)"
-mkdir -p "${TMPDIR}"
-cd "${TMPDIR}"
-
-for VERSION in $(yq e /deckhouse/candi/version_map.yml -j | jq -r '.k8s | keys[]'); do
-  for PATCH in $(yq e /deckhouse/candi/version_map.yml -j | jq -r --arg version "${VERSION}" '.k8s."\($version)".patch'); do
+for VERSION in $(yq e ../version_map.yml -o json | jq -r '.k8s | keys[]'); do
+  for PATCH in $(yq e ../version_map.yml -o json | jq -r --arg version "${VERSION}" '.k8s."\($version)".patch'); do
     # Get last patch version from github CHANGELOG.md
     NEW_FULL_VERSION="$(curl -s "https://raw.githubusercontent.com/kubernetes/kubernetes/master/CHANGELOG/CHANGELOG-${VERSION}.md" | grep '## Downloads for v' | head -n 1 | grep -Eo "${VERSION}.[0-9]+")"
     NEW_PATCH="$(awk -F "." '{print $3}' <<< "${NEW_FULL_VERSION}")"
     if [[ "${NEW_PATCH}" -ne "${PATCH}" ]]; then
-      echo "New kubernetes patch version ${NEW_PATCH} found for ${VERSION}"
+      PR_DESCRIPTION="${PR_DESCRIPTION}$(echo -e "\n* New kubernetes patch version ${VERSION}.${NEW_PATCH}.")"
+      CREATE_PR=true
       update_version_map "${VERSION}" "${NEW_PATCH}"
     fi
   done
 done
 
-cd -
+if [[ "${CREATE_PR}" == "true" ]]; then
+  cd ../../
+  BRANCH="kubernetes-patchversions-$(date +"%y-%m-%d-%H-%M")"
+  git checkout -b "${BRANCH}"
+  git add .
```

Comment:
It seems like `git add .` can accidentally add more files than needed. If we know the exact files changed, it is better to add only these files
```suggestion
  git add ../version_map.yml
```